library/src/main/java/io/ona/kujaku/activities/MapActivity.java

```diff
@@ -101,79 +102,95 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_map);
         initializeViews();
 
-        checkPermissions();
         screenWidth = getScreenWidth(this);
 
+        Bundle bundle = getIntentExtras();
+        if (bundle != null) {
+            String mapBoxAccessToken = bundle.getString(Constants.PARCELABLE_KEY_MAPBOX_ACCESS_TOKEN);
+            Mapbox.getInstance(this, mapBoxAccessToken);
+        }
+
+        checkPermissions(savedInstanceState);
+    }
+
+    private Bundle getIntentExtras() {
         if (getIntent() != null) {
             Bundle bundle = getIntent().getExtras();
-            if (bundle != null){
-                String mapBoxAccessToken = bundle.getString(Constants.PARCELABLE_KEY_MAPBOX_ACCESS_TOKEN);
-                String[] stylesArray = bundle.getStringArray(Constants.PARCELABLE_KEY_MAPBOX_STYLES);
-                currentStylePath = "";
+            if (bundle != null) {
+                return bundle;
+            }
+        }
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_TOP_LEFT_BOUND)) {
-                    topLeftBound = bundle.getParcelable(Constants.PARCELABLE_KEY_TOP_LEFT_BOUND);
+        return null;
+    }
 
-                    if (bundle.containsKey(Constants.PARCELABLE_KEY_BOTTOM_RIGHT_BOUND)) {
-                        bottomRightBound = bundle.getParcelable(Constants.PARCELABLE_KEY_BOTTOM_RIGHT_BOUND);
-                    }
-                }
+    private void initializeMapActivityAfterPermissionsSupplied(Bundle savedInstanceState) {
+        Bundle bundle = getIntentExtras();
+        if (bundle != null) {
+            String[] stylesArray = bundle.getStringArray(Constants.PARCELABLE_KEY_MAPBOX_STYLES);
+            currentStylePath = "";
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_MAX_ZOOM)) {
-                    maxZoom = bundle.getDouble(Constants.PARCELABLE_KEY_MAX_ZOOM);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_TOP_LEFT_BOUND)) {
+                topLeftBound = bundle.getParcelable(Constants.PARCELABLE_KEY_TOP_LEFT_BOUND);
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_MIN_ZOOM)) {
-                    minZoom = bundle.getDouble(Constants.PARCELABLE_KEY_MIN_ZOOM);
+                if (bundle.containsKey(Constants.PARCELABLE_KEY_BOTTOM_RIGHT_BOUND)) {
+                    bottomRightBound = bundle.getParcelable(Constants.PARCELABLE_KEY_BOTTOM_RIGHT_BOUND);
```

Comment:
Can we not use MapBox's bounds for the vector layer? https://www.mapbox.com/mapbox-gl-js/style-spec/#sources-vector-bounds

Reply:
I think vector layer zoom constraints and map zoom constraints are different

Reply:
Going to create an issue for this.